CPAC/seg_preproc/utils.py

```diff
@@ -319,3 +319,168 @@ def erosion(roi_mask = None, erosion_mm = None, erosion_prop = None):
+        err_msg = '\n\n[!] C-PAC says: '\
+                    'Please check ANTs Prior-based Segmentation setting. ' \
+                    'For performing ANTs Prior-based segmentation method '\
+                    'the number of specified segmentations should be identical to the number of atlas image sets.'\
+                    '\n\n'
+        raise Exception(err_msg)    
+    else:
+        for index in range(len(template_brain_list)):
+            cmd.append(" -g {0} -l {1}".format(template_brain_list[index], template_segmentation_list[index]))
+
+    # write out the actual command-line entry for testing/validation later
+    command_file = os.path.join(os.getcwd(), 'command.txt')
+    with open(command_file, 'wt') as f:
+        f.write(' '.join(cmd))
+    
+    str = ""
+    bash_cmd = str.join(cmd) 
+
+    try:
+        retcode = subprocess.check_output(bash_cmd, shell=True) 
+    except Exception as e:
+        raise Exception('[!] antsJointLabel segmentation method did not complete successfully.'
+                        '\n\nError details:\n{0}\n{1}\n'.format(e, e.output))
+
+    multiatlas_Intensity = None
+    multiatlas_Labels = None
+
+    files = [f for f in os.listdir('.') if os.path.isfile(f)]
+
+    for f in files:
+        if "Intensity" in f:
+            multiatlas_Intensity = os.getcwd() + "/" + f
+        if "Labels" in f:
+            multiatlas_Labels = os.getcwd() + "/" + f
+
+    if not multiatlas_Labels:
+        raise Exception("\n\n[!] No multiatlas labels file found."
+                        "antsJointLabelFusion may not have completed successfully.\n\n")
+
+    return multiatlas_Intensity, multiatlas_Labels
+
+
+def pick_tissue_from_labels_file(multiatlas_Labels, csf_label, 
+                                left_gm_label, left_wm_label,
+                                right_gm_label, right_wm_label):
+
+
+    """
+    pick tissue mask from multiatlas labels file
+    https://surfer.nmr.mgh.harvard.edu/fswiki/FsTutorial/AnatomicalROI/FreeSurferColorLUT
+
+    Parameters
+    ----------
+
+    multiatlas_Labels : string (nifti file)
+
+    csf_label: integer 
+        label value corresponding to CSF in multiatlas file
+
+    left_gm_label: integer 
+        label value corresponding to Left Gray Matter in multiatlas file
+
+    left_wm_label: integer 
+        label value corresponding to Left White Matter in multiatlas file
+
+    right_gm_label: integer 
+        label value corresponding to Right Gray Matter in multiatlas file
+
+    right_wm_label: integer 
+        label value corresponding to Right White Matter in multiatlas file
+
+    Returns
+    -------
+
+    csf_mask : string (nifti file)
+
+    gm_mask : string (nifti file)
+
+    wm_mask : string (nifti file)
+
+    """
+    import os
+    import nibabel as nb
+    import numpy as np
+
+    img = nb.load(multiatlas_Labels)
+    data = img.get_data()
+
+    csf = data.copy()
+    if csf_label == None:
+        csf[csf != 24] = 0
```

Comment:
I think these values need to be tied in from the config file?

Reply:
yep. it is actually tied to the values in config file. [https://github.com/FCP-INDI/C-PAC/blob/feature/antsprior_seg/CPAC/seg_preproc/utils.py#L450-L455](url). I leave a None just in case csf/gm/wm labels are not specified. then in this case, will keep csf/gm/wm labels value same as FreeSurferColorLUT. [https://github.com/FCP-INDI/C-PAC/blob/feature/antsprior_seg/dev/docker_data/default_pipeline.yml#L428-L445](url)

Reply:
Ah yeah, my fault, I kind of glossed over the == None. I would say just drop in a comment in the code mentioning that those hard-coded values are based off of FreeSurferColorLUT.